src/index.ts

```diff
@@ -1,151 +1,89 @@
     return pack.version
   }
 
-  private socket: WebSocket | undefined
-  private requests: Record<string, (res: Message) => void> = {}
-  private subscriptions: Record<string, Subscription> = {}
-
-  public async connect(url: string) {
-    return new Promise((resolve, reject) => {
-      this.socket = new WebSocket(url) as WebSocket
+  private host: string | undefined
+  // private subscriptions: Record<string, Subscription> = {}
 
-      this.socket.onopen = () => {
-        console.debug('Connection opened.')
-        resolve()
-      }
-      this.socket.onclose = () => {
-        console.debug('Connection closed.')
-      }
-      this.socket.onerror = (e) => {
-        console.error(e)
-        reject(e)
-      }
-      this.socket.onmessage = (e) => {
-        this.handleEvent(e)
-      }
-    })
+  constructor() {
+    grpc.setDefaultTransport(grpc.FetchReadableStreamTransport({
+      credentials: 'omit'
+    }))
+    // grpc.setDefaultTransport(grpc.WebsocketTransport())
   }
 
-  public async addThread(addr: string, followKey: string, readKey: string) {
-    return this.send('addThread', [addr, followKey, readKey])
+  public setHost(host: string) {
+    this.host = host
+    return this
   }
 
-  public async pullThread(id: string) {
-    return this.send('pullThread', [id])
+  public async newStore() {
+    return this.unary(API.NewStore, new NewStoreRequest())
   }
 
-  public async deleteThread(id: string) {
-    return this.send('deleteThread', [id])
+  public async registerSchema(storeID: string, name: string, schema: any) {
+    const req = new RegisterSchemaRequest()
+    req.setStoreid(storeID)
+    req.setName(name)
+    req.setSchema(JSON.stringify(schema))
+    return this.unary(API.RegisterSchema, req)
   }
 
-  public async addFollower(addr: string) {
-    return this.send('addFollower', [addr])
-  }
+  public async modelCreate(storeID: string, modelName: string, values: any[]) {
+    const req = new ModelCreateRequest()
+    req.setStoreid(storeID)
+    req.setModelname(modelName)
 
-  public async addRecord(body: any, threadID: string) {
-    return this.send('addRecord', [JSON.stringify(body), threadID])
-  }
-
-  public async getRecord(recordID: string, threadID: string) {
-    return this.send('getRecord', [threadID, recordID])
-  }
-
-  public async subscribe(threads: string[], handler: (rec: LogRecord) => void) {
-    return this.send('subscribe', threads).then(() => {
-      const token = uuid.v4()
-      this.subscriptions[token] = {
-        threads,
-        handler
-      }
-      return token
+    const list: any[] = []
+    values.forEach((v) => {
+      v['ID'] = uuid.v4()
+      list.push(JSON.stringify(v))
     })
+    req.setValuesList(list)
+
+    return this.unary(API.ModelCreate, req)
   }
 
-  protected async send(method: string, args: string[]) {
+  public async unary<TRequest extends ProtobufMessage, TResponse extends ProtobufMessage, M extends UnaryMethodDefinition<TRequest, TResponse>>(
```

Comment:
Should this be protected or private? Just feels so low level that maybe we don't want to expose?

Reply:
Oh yes, good call.